webforces/server/mongodbworker.py

```diff
@@ -12,23 +13,208 @@ def __init__(self) -> None:
+        try:
+            id_collection = self.db["id"]
+            dict = id_collection.find_one({"name": name})
+            new_id = dict["last_id"] + 1
+            id_collection.update_one(
+                {"name": name},
+                {"$set": {"last_id": new_id}}
+            )
+        except Exception as e:
+            logger.error(f"MongoDBWorker connection failed: {e}")
+            return DBStatuses.s_data_issue
+        return new_id
+
+    def _insertOneID(self, name) -> int:
+        try:
+            id_collection = self.db["id"]
+            id_collection.insert_one(
+                {"name": name, "last_id": 0}
+            )
+        except Exception as e:
+            logger.error(f"MongoDBWorker connection failed: {e}")
+            return DBStatuses.s_data_issue
+        return DBStatuses.s_ok
+
+    def addUser(self, user) -> (int, User):
+        try:
+            users_collection = self.db["users"]
+            if users_collection.find_one({"login": user.login}) is not None:
+                logger.error("This login is already taken")
+                return (DBStatuses.s_data_issue, User(-100))
+            user.user_id = self._getNextID("users")
+            if user.user_id == DBStatuses.s_data_issue:
+                return (DBStatuses.s_connection_error, User(-100))
+            users_collection.insert_one(user.__dict__)
+            if self._insertOneID(str(user.user_id) + "u") == DBStatuses.s_data_issue:
+                return (DBStatuses.s_connection_error, User(-100))
+        except Exception as e:
+            logger.error(f"MongoDBWorker connection failed: {e}")
+            return (DBStatuses.s_connection_error, User(-100))
+        logger.debug("New user was successfully added")
+        return (DBStatuses.s_ok, user)
+
+    def getUserByID(self, user_id) -> (int, User):
+        try:
+            users_collection = self.db["users"]
+            user_d = users_collection.find_one({"user_id": user_id})
+            if user_d is None:
+                logger.error("This user does not exist")
+                return (DBStatuses.s_data_issue, User(-100))
+        except Exception as e:
+            logger.error(f"MongoDBWorker connection failed: {e}")
+            return (DBStatuses.s_connection_error, User(-100))
+        logger.debug("User was found")
+        return (DBStatuses.s_ok, User.fromDict(user_d))
+
+    def getUserByLogin(self, login) -> (int, User):
+        try:
+            users_collection = self.db["users"]
+            user_d = users_collection.find_one({"login": login})
+            if user_d is None:
+                logger.error("This user does not exist")
+                return (DBStatuses.s_data_issue, User(-100))
+        except Exception as e:
+            logger.error(f"MongoDBWorker connection failed: {e}")
+            return (DBStatuses.s_connection_error, User(-100))
+        logger.debug("User was found")
+        return (DBStatuses.s_ok, User.fromDict(user_d))
+
+# TODO
```

Comment:
These functions will be fixed in the next PR and we can merge these for now, am I right?

Reply:
Yes.